fusilli-plugin/docs/TheRockRFC.md

```diff
@@ -0,0 +1,34 @@
+| author | created | modified | status |
+|--------|---------|----------|--------|
+| Sambhav Jain, Aaron St. George and Mahesh Ravishankar | <TBD> | <TBD> | draft |
+
+
+# Add IREE as a kernel provider for hipDNN
+
+## Overview
+
+[IREE](https://github.com/iree-org/iree/) is an open source ML compiler stack built using MLIR that is intended to support the compilation and execution of ML models. While IREE is setup to be multi-targeting, over the past couple of years a lot of effort has gone into improving the codegeneration for AMDGPUs, specifically Instinct class GPUs. While a lot of the IREE compiler stack is meant to optimize execution of full-scale ML models, one key component of the work is to have efficient kernel code generation for MI300+ cards. [Fusilli](https://github.com/nod-ai/shark-ai/tree/main/sharkfuser) is a project built on top of IREE that leverages the kernel codegeneration capabilities of IREE and packages it to be useable as a kernel provider within hipDNN. The advantages of using IREE this way are
+
+1) IREE has been built from the ground-up as a fusion compiler. The kinds of fusions that libraries like hipDNN are expected to provide are supported out-of-the box in IREE. 
+2) Being a compiler approach, IREE as a kernel provided functions as a JIT-backend. For cases where IREE generated code is faster, IREE would JIT-compile the kernels without having to pre-build kernels and ship those with hipDNN.
+
+This RFC is to propose adding a path to using IREE as a kernel provider to hipDNN within TheRock. There are three components to reason about
+
+1. The hipDNN backend plugin to Fusilli. Currently it lives [here](https://github.com/nod-ai/shark-ai/tree/main/fusilli-plugin)
+2. Fusilli. Currently lives [here](https://github.com/nod-ai/shark-ai/tree/main/sharkfuser)
+3. IREE, which lives in its own Github Org and is a Linux Foundation Project. It currently lives [here](https://github.com/iree-org/iree)
+
+## Workplan
+
+### Immediate next steps
+
+The immediate workplan is to move just the hipDNN backend plugin to Fusilli (so component 1 above) into TheRock. The plugin will be built conditionally (not on by default) and will pull in Fusilli as an external dependency. Nested dependencies of Fusilli are (TODO: Adjust/clarify the following).
+
+1. IREE runtime sources are built into Fusilli
```

Comment:
This is not entirely true. Fusilli standalone doesn't build IREE runtime - this is brought in pre-built. The plugin when built does statically link in both fusilli and IREE runtime into the final .so.

Reply:
Sure, let's land this and we can iterate on it, given I can't push to your fork.